packages/next/src/utils.ts

```diff
@@ -1436,16 +1438,36 @@ export async function getPageLambdaGroups({
 
     let opts: { memory?: number; maxDuration?: number } = {};
 
+    if (functionsConfigManifest && functionsConfigManifest[routeName]) {
+      opts = functionsConfigManifest[routeName];
+    }
+
     if (config && config.functions) {
       const sourceFile = await getSourceFilePathFromPage({
         workPath: entryPath,
         page,
         pageExtensions,
       });
-      opts = await getLambdaOptionsFromFunction({
+
+      const configOpts = await getLambdaOptionsFromFunction({
         sourceFile,
         config,
       });
+
+      // If a user tries to customize in both handler and vercel.json we throw
+      const duplicateKeys = Object.keys(opts).filter(key =>
+        Object.keys(configOpts).includes(key)
+      );
+
+      if (duplicateKeys.length > 0) {
+        throw new Error(
+          `The route ${routeName} has duplicate configuration in both handler and vercel.json for: ${duplicateKeys.join(
```

Comment:
Please add a test for this case

Reply:
Good catch, actually removing this as currently the default behavior is that `vercel.json` takes precedent.